src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs

```diff
@@ -77,19 +77,28 @@ public EditorConfigStorageLocation(string keyName, Func<string, object> parseVal
         public EditorConfigStorageLocation()
         {
             // If the user didn't pass a keyName assume we need to parse the entire dictionary
-            _tryParseDictionary = (dictionary, type) =>
+            _tryParseDictionary = (dictionary, type, underlyingOption) =>
             {
                 if (type == typeof(NamingStylePreferences))
                 {
-                    var result = EditorConfigNamingStyleParser.GetNamingStylesFromDictionary(dictionary);
-                    if (!result.NamingRules.Any() &&
-                        !result.NamingStyles.Any() &&
-                        !result.SymbolSpecifications.Any())
+                    var editorconfigNamingStylePreferences = EditorConfigNamingStyleParser.GetNamingStylesFromDictionary(dictionary);
+                    
+                    if (!editorconfigNamingStylePreferences.NamingRules.Any() &&
+                        !editorconfigNamingStylePreferences.NamingStyles.Any() &&
+                        !editorconfigNamingStylePreferences.SymbolSpecifications.Any())
                     {
-                        return (result: result, succeeded: false);
+                        // We were not able to parse any rules from editorconfig, tell the caller that the parse failed
+                        return (result: editorconfigNamingStylePreferences, succeeded: false);
```

Comment:
returning false here will cause us to fall back to the workspace options as seen [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Options/OptionServiceFactory.cs,191)